### PR TITLE
Remove Clang+CUDA workaround

### PR DESCRIPTION
**Pull request by dalg24**
Does not seem to be needed any more.

### CODE REVIEW

File: core/unit_test/TestTeamVector.hpp
Comment (masterleinad) [inline core/unit_test/TestTeamVector.hpp:1]: This worked fine the last time I tested it with recent `clang++`.

Comment (dalg24) [reply core/unit_test/TestTeamVector.hpp:1]: I don't have a Clang10 handy to check unfortunately
I would love to get rid of this macro

**Comment by ndellingwood** [comment]
No regressions in local testing of this PR using clang/10 with cuda/10 on Volta70

**Review by masterleinad — APPROVED** [review]

**Comment by dalg24** [comment]
Unrelated OpenMPTarget failure
```
48: [ RUN      ] simd.device_math_ops
48: SIMD unit test equality condition failed on deviceSIMD unit test equality condition failed on deviceSIMD unit test equality condition failed on deviceSIMD unit test equality condition failed on deviceSIMD unit test equality condition failed on deviceSIMD unit test equality condition failed on device"PluginInterface" error: Failure to synchronize stream (nil): Error in cuStreamSynchronize: an illegal memory access was encountered
48: Libomptarget error: Consult https://openmp.llvm.org/design/Runtimes.html for debugging options.
48: Kokkos_OpenMPTarget_ParallelFor_Range.hpp:54:1: Libomptarget fatal error 1: failure of target construct while offloading is mandatory
48/48 Test #48: Kokkos_UnitTest_SIMD ........................Subprocess aborted***Exception:   3.52 sec
```
being fixed in #

**Review by fnrizzi — APPROVED** [review]

**Comment by fnrizzi** [comment]
> being fixed in #

missing some number here or reference to something?

**Comment by dalg24** [comment]
> > being fixed in #
> 
> missing some number here or reference to something?

I meant to remove the sentence.
I was going to reference #6574 and then realize it was disabling not fixing.

**Pull request merged**